kyo-core/js/src/main/scala/kyo/KyoAppPlatformSpecific.scala

```diff
@@ -0,0 +1,30 @@
+package kyo
+
+import kyo.Maybe.Absent
+import kyo.Maybe.Present
+
+abstract class KyoAppPlatformSpecific extends KyoApp.Base[Async & Resource & Abort[Throwable]]:
+
+    private var maybePreviousAsync: Maybe[Unit < (Async & Abort[Throwable])] = Absent
+
+    final override protected def run[A: Flat](v: => A < (Async & Resource & Abort[Throwable]))(using Frame): Unit =
+        import AllowUnsafe.embrace.danger
+        val currentAsync: Unit < (Async & Abort[Throwable]) =
+            Async.timeout(timeout)(handle(v)).map(value => printResult(Result.success(value)))
```

Comment:
If you put the `timeout` within the initiation of the Unit function, (when you update `initCode`) you will avoid some sleep fibers and potentially racy behavior.

I think it makes sense to timeout the whole application anyways.

Reply:
I mimiced the code in JVM (timeout for each `run`s). Will updated this